cmds/legacy_boot/legacy_boot.go

```diff
@@ -0,0 +1,383 @@
+	files, err := ioutil.ReadDir(blkpath)
+	if err != nil {
+		log.Fatal(err)
+	}
+	for _, file := range files {
+		device_entry, err := ioutil.ReadDir(blkpath + file.Name() + devpath)
+		if err == nil {
+			blk_devices = append(blk_devices, device_entry[0].Name())
+		}
+	}
+	return (blk_devices)
+}
+
+// Current support is limited to Hard disk devices and USB devices
+
+func check_for_bootable_mbr(path string) int {
+	var b511, b510 byte
+	f, err := os.Open(path)
+	check(err)
+	b1 := make([]byte, 512)
+	f.Read(b1)
+	b511 = b1[511]
+	b510 = b1[510]
+	f.Close()
+	if ((b511 == 0xaa) && (b510 == 0x55)) == true {
+		return 1
+	}
+	return 0
+}
+
+// Return all devices attached to a specific name like /dev/sdaX where X can move from 0 to 127
+// FIXME no support for devices which are included into subdirectory within /dev
+
+func get_device_part_list(path string) []string {
+	var return_value []string
+	files, err := ioutil.ReadDir("/dev/")
+	if err != nil {
+		log.Fatal(err)
+	}
+	for _, file := range files {
+		if strings.HasPrefix(file.Name(), path) {
+			// We shall not return full device name
+			if file.Name() != path {
+				// We shall check that the remaining part is a number
+				return_value = append(return_value, file.Name())
+			}
+		}
+	}
+	return return_value
+}
+
+// Return all block file system supported by the linuxboot kernel
+
+func get_supported_filesystem() []string {
+	var return_value []string
+	file, err := os.Open("/proc/filesystems")
+	check(err)
+	defer file.Close()
+	scanner := bufio.NewScanner(file)
+	if err := scanner.Err(); err != nil {
+		log.Fatal(err)
+	}
+	var string1, string2 string
+	for scanner.Scan() {
+		string1 = ""
+		string2 = ""
+		fmt.Sscanf(scanner.Text(), "%s %s", &string1, &string2)
+		if string2 == "" {
+			if string1 != "" {
+				return_value = append(return_value, string1)
+			}
+		}
+	}
+	return return_value
+
+}
+
+// try to mount a specific block device
+
+func mount_entry(path string, supported_filesystem []string) bool {
+	var return_value bool
+	syscall.Mkdir("/u-root", 0777)
+	var flags uintptr
+	// Was supposed to be unecessary for kernel 4.x.x
+	if verbose {
+		println("/dev/" + path)
+	}
+	for _, filesystem := range supported_filesystem {
+		flags = syscall.MS_MGC_VAL
+		// Need to load the filesystem kind supported
+		syscall.Mkdir("/u-root/"+path, 0777)
+		err := syscall.Mount("/dev/"+path, "/u-root/"+path, filesystem, flags, "")
+		if err == nil {
+			return true
+		}
+	}
+	return_value = false
+	return return_value
+}
+
+func umount_entry(path string) bool {
+	var return_value bool
+	var flags int
+	// Was supposed to be unecessary for kernel 4.x.x
+	flags = syscall.MNT_DETACH
+	err := syscall.Unmount(path, flags)
+	if err == nil {
+		return true
+	}
+	return_value = false
+	return return_value
+}
+
+// This function is looking for grub.cfg file
+// and return absolute path to it
+
+func check_boot_entry(mount_point string) string {
+	_, err := os.Stat(mount_point + "/boot")
```

Comment:
Why not just os.Stat(filepath.Join(mountPoint, "boot", "grub", "grub.cfg"))? No need to stat the parents.